### PR TITLE
feat(storage): reduce cost of metrics and stack trace

### PR DESCRIPTION
**Pull request by Little-Wallace**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

see CPU flame graph in https://buildkite.com/organizations/risingwave-test/pipelines/tpch-benchmark/builds/1010/jobs/018df08c-e78f-4aae-89c8-0d7cf40531cd/artifacts/018df0b5-15b7-43e0-8238-f1b5e300a3bb

The metrics will cost cpu if we query the labels every time calling `iter`

Link #15478 

## Checklist

- [x] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Review by MrCroxx — APPROVED** [review]
LGTM

Just a notice: TRACE level span won't be compiled in release mode. It's okay to keep them. But if they are verbose, remove them as you wish.

File: src/storage/src/hummock/sstable_store.rs [hunk 1]
```diff
@@ -655,7 +655,7 @@ impl SstableStore {
             _ => (),
         }
         stats.cache_meta_block_total += 1;
-        lookup_response.verbose_instrument_await("sstable")
+        lookup_response
```
Comment (BugenZhao) [inline src/storage/src/hummock/sstable_store.rs:658]: Is there any indication that `await-tree` results in high CPU overhead?

Comment (Little-Wallace) [reply src/storage/src/hummock/sstable_store.rs:658]: Yes. A little... we tested it but i can not found the flamegraph now ...

Comment (BugenZhao) [reply src/storage/src/hummock/sstable_store.rs:658]: I believe there's always a trade-off between observability and performance. 🤣 If we find the impact little, perhaps we should keep it untouched.

**Comment by Little-Wallace** [comment]
> LGTM
> 
> Just a notice: TRACE level span won't be compiled in release mode. It's okay to keep them. But if they are verbose, remove them as you wish.

No... It seems work actually in release mode...

**Comment by BugenZhao** [comment]
> > LGTM
> > Just a notice: TRACE level span won't be compiled in release mode. It's okay to keep them. But if they are verbose, remove them as you wish.
> 
> No... It seems work actually in release mode...

That shouldn't be possible. By release I think it's referring to the optimized build without debug assertions, which does not apply to `ci-release` though.

**Pull request merged**